### PR TITLE
Demo: gradient descent

### PR DESCRIPTION
**Pull request by blide**

### CODE REVIEW

File: demos/GradientDescent.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+// path/to/libclad.so  -I../include/ -x c++ -lstdc++ -lm GradientDescent.cpp
+//
+// A typical invocation would be:
+// ../../../../obj/Debug+Asserts/bin/clang  -Xclang -add-plugin -Xclang clad \
+// -Xclang -load -Xclang ../../../../obj/Debug+Asserts/lib/libclad.dylib     \
+// -I../include/ -x c++ -lstdc++ -lm GradientDescent.cpp
+//
+
+#include <iostream>
+#include "math.h"
```
Comment (vgvassilev) [inline demos/GradientDescent.cpp:19]: Why not including cmath?

File: demos/GradientDescent.cpp [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+typedef void (*targetGradT) (double, double, double, double*);
+
+class GradientDescent {
+public:
+  GradientDescent(targetT target, targetGradT targetGrad):
+  target(target), targetGrad(targetGrad){
+    setParams(Point3D(0, 0, 0));
+  }
+
+  bool isConverged() {
```
Comment (vgvassilev) [inline demos/GradientDescent.cpp:42]: Did you mean `hasConverged`?

File: demos/GradientDescent.cpp [hunk 3]
```diff
@@ -0,0 +1,120 @@
+//--------------------------------------------------------------------*- C++ -*-
+// clad - The C++ Clang-based Automatic Differentiator
+//
+// Gradient descent demo.
```
Comment (vgvassilev) [inline demos/GradientDescent.cpp:4]: Can you explain a little more here of the intent of the demo?

**Comment by vgvassilev** [comment]
@blide, can you resolve the conflict?

**Comment by blide** [comment]
@vgvassilev, done.

**Comment by vgvassilev** [comment]
PR #191 was inspired by this example. I am closing it now as the work is done.

**Pull request closed without merging**